generators/typegen/generator.go

```diff
@@ -126,15 +158,57 @@ func (g *Generator) generateTypesForPackage(s *schema.Schema, genConfig *config.
 			}
 
 			enumsForGen = append(enumsForGen, xxx)
+		case schema.KindScalar:
+			log.Warnf("scalar type: %+v", t)
+
+			// Default scalars to int
+			createAs := "int"
+			skipTypeCreate := false
+
+			for _, p := range pkgConfig.Types {
```

Comment:
There is no break from this loop, so what happens if `p.Name` matches more than once with different createAs/skipTypeCreate values?

Reply:
Duplicate name detection could make sense, but perhaps not here.

Reply:
I was thinking a `break` when found, at least that would be 'first wins' vs a possible mixture of more than one match. (i.e. first match `createAs` is set and overrides, and second match `skipTypeCreate` is set)

Reply:
I've included a little handling for this now.